File: continuous-subarray-sum.rs
use std::collections::HashSet;

impl Solution {
  pub fn check_subarray_sum(nums: Vec<i32>, k: i32) -> bool {
    let mut s = HashSet::new();
    let (mut l1, mut run) = (0, 0);
    for num in nums {
      run = (run + num) % k;
      if s.contains(&run) {
        return true;
      }

      s.insert(l1);
      l1 = run;
    }
    false
  }
}

File: c/continuous-subarray-sum.cc
using ll = long long;

class Solution {
public:
  bool checkSubarraySum(vector<int>& nums, int k) {
    map<ll, int> t;
    ll l1 = 0, run = 0;
    for (int e : nums) {
      run = (run + e) % k;
      if (t.count(run)) return true;

      t[l1] = 1;
      l1 = run;
    }
    return false;
  }
};
